### PR TITLE
[Impeller] Don't use waitUntilScheduled on iOS

### PR DESCRIPTION
**Pull request by bdero**

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]

**Review by dnfield — CHANGES_REQUESTED** [review]
Want to make sure this doesn't land without getting some more context in the code this time.

File: impeller/renderer/backend/metal/command_buffer_mtl.mm [hunk 1]
```diff
@@ -166,7 +166,9 @@ static bool LogMTLCommandBufferErrorIfPresent(id<MTLCommandBuffer> buffer) {
   }
 
   [buffer_ commit];
+#if (FML_OS_MACOSX || FML_OS_IOS_SIMULATOR)
```
[the remark this answers is not in the document]
Comment (bdero) [reply impeller/renderer/backend/metal/command_buffer_mtl.mm:170]: Alrighty, done.

[the remark this answers is not in the document]
Comment (dnfield) [reply impeller/renderer/backend/metal/command_buffer_mtl.mm:170]: Are there any instructions on how to reproduce the issue seen? E.g. `Run wonderous app on a simulator on an arm mac host`?

[the remark this answers is not in the document]
Comment (chinmaygarde) [reply impeller/renderer/backend/metal/command_buffer_mtl.mm:170]: Yeah, thats pretty much exactly it.

**Review by dnfield — APPROVED** [review]

**Pull request merged**